Assets/MixedRealityToolkit/Utilities/CameraCache.cs

```diff
@@ -1,49 +1,86 @@
-                    {   // If the cached camera is active, return it
-                        // Otherwise, our playspace may have been disabled
-                        // We'll have to search for the next available
-                        return cachedCamera;
-                    }
+                    return cameraService.Main;
                 }
 
-                // If the cached camera is null, search for main
-                var mainCamera = Camera.main;
+                // Otherwise, return a default camera
+                return FindOrCreateDefaultMainCamera();
+            }
+        }
 
-                if (mainCamera == null)
-                {   // If no main camera was found, create it now
-                    Debug.LogWarning("No main camera found. The Mixed Reality Toolkit requires at least one camera in the scene. One will be generated now.");
-                    mainCamera = new GameObject("Main Camera", typeof(Camera)) { tag = "MainCamera" }.GetComponent<Camera>();
+        /// <summary>
+        /// Returns true if cached camera exists and is enabled.
+        /// </summary>
+        public static bool MainExists
+        {
+            get
+            {
+                if (MixedRealityServiceRegistry.TryGetService<IMixedRealityCameraSystem>(out cameraService))
+                {   // The camera system is guaranteed to create a camera.
+                    return true;
                 }
 
-                // Cache the main camera
-                cachedCamera = mainCamera;
+                return cachedCamera != null && cachedCamera.isActiveAndEnabled;
+            }
+        }
 
+        /// <summary>
+        /// Creates a default main camera if our cached camera is null or disabled.
+        /// </summary>
+        /// <returns></returns>
+        private static Camera FindOrCreateDefaultMainCamera()
+        {
+            if (cachedCamera != null && cachedCamera.isActiveAndEnabled)
+            {
+                // If the cached camera is active, return it
+                // Otherwise, our playspace may have been disabled
+                // We'll have to search for the next available
                 return cachedCamera;
             }
+
+            // If the cached camera is null, search for main
+            var mainCamera = Camera.main;
+
+            if (mainCamera == null || !mainCamera.isActiveAndEnabled)
+            {   // If no main camera was found, create it now
+                Debug.LogWarning("No main camera found. The Mixed Reality Toolkit requires at least one camera in the scene. One will be generated now.");
+                mainCamera = new GameObject("Main Camera", typeof(Camera)) { tag = "MainCamera" }.GetComponent<Camera>();
+            }
+
+            // Cache the main camera
+            cachedCamera = mainCamera;
+            // Parent the camera under our play space
+            cachedCamera.transform.SetParent(MixedRealityPlayspace.Transform);
```

Comment:
Won't this already be set, if needed, in the `Transform` getter? https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4455/files#diff-a2a2bab0eb44c8a46d3c07c490881396R214

Reply:
It only parents the main camera if no camera system is found. The camera cache doesn't assume that a main camera delivered by the service belongs under the playspace transform.

In our default implementation we do parent the camera under the playspace, but custom implementations may do something different.